### ISSUE TITLE
loss become nan during training

### ISSUE THREAD

**Issue by FANzhaoxin666** [jac99/MinkLocMultimodal#11]
hello!
 I run your code and find the loss becomes nan after 40 epoch. I find it is caused by that values of all embeddings become nan.  Do you find this during your research? Is there any bug? or just by accident?

**Comment by jac99** [comment]
Hi,
I haven't encountered this in my experiments. Can you post details of _config_ and _model_config_ files, used for the training? Can you also post a screenshot of the output from the training process, just before a loss becomes nan (e.g. screenshot showing output for epochs 35 till 42 or similar).

**Comment by FANzhaoxin666** [comment]
> Hi, I haven't encountered this in my experiments. Can you post details of _config_ and _model_config_ files, used for the training? Can you also post a screenshot of the output from the training process, just before a loss becomes nan (e.g. screenshot showing output for epochs 35 till 42 or similar).

Thank you. Sorry for that I didn't save the log. The config and model_config are the same with you. We retrained the model for several times. We find it is the image_loss become nan. And the total loss kept to be 0.2. We also find that there is no batch zise expanding during training. We suspect it may be caused by pytorch version. We use pytorch 1.8 but we notice you use 1.9.1. Before using pytorch 1.8, we also tried to use pytorch 1.7, but there would be an error: the positive/negative mask value must be bool type. Therefore, we think   it is the different pytorch version try the positive/negative mask differently caused the problem.  We are now trying to replace the pytorch version to 1.9 or 1.10. It there any new findings, we will tell you. Thanks.

**Comment by FANzhaoxin666** [comment]
![image](https://user-images.githubusercontent.com/45939914/143772382-33de6708-07ac-46a7-9659-08b4df40eaad.png)
Here is my config.

**Comment by FANzhaoxin666** [comment]
![9bd8b28ac846a0534cdb34650d9616d](https://user-images.githubusercontent.com/45939914/143772424-7ef6701c-acaf-4f26-8c72-cc7827af1bfc.png)
He is a screen shot of the newest training we tried.

**Comment by jac99** [comment]
If the loss is close to the margin (0.2) and the number of active triplets is equal to the initial batch size and does not decrease (initial batch size = 8 and the number of active triplets is 8) this indicates some problem in the training process. Likely all embeddings collapse to zero and the network is not training. After a few epochs, all embeddings become zero and this causes NaNs.
The expected behaviour is that the number of active triplets starts decreasing from epoch 1. Then, when it decreases below a threshold (when there's less than 70% of active triplets), the batch size is automatically increased (until it reaches batch_size_limit=160). 
I suggest to try training the unimodal model first (point cloud only model): 
`python train.py --config ../config/config_baseline.txt --model_config ../models/minkloc3d.txt`
If it trains correctly, and evaluation at the end of the training gives expected results, it means there's likely some issue with training images.
I'll try to clone the github repo later and run the training again, to verify that the github code works correctly.

**Comment by jac99** [comment]
Also ensure that you are using the latest version of the code. Some time ago there were some changes done in the repo, and older versions (like 3 months old or so) may not work correctly.

**Comment by FANzhaoxin666** [comment]
Thank you! The problem is solved by replacing the code you released 2 months ago with the latest version. Thank you so much.

**Comment by jac99** [comment]
Issue is solved.

**Comment by jac99** [comment]
Issue is solved, closing.
